bitlbee-facebook: Add agent version bump patch

toonn opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

The plugin has recently stopped working. The patch was very simple but hasn't made it into a release yet. Without it the plugin is fairly useless currently.

I have been running the master branch of bitlbee-facebook, which includes the commit this patch adds, for a couple days and haven't experienced any problems.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

toonn committed:
bitlbee-facebook: Add agent version bump patch

The plugin has recently stopped working. The patch was very simple but
hasn't made it into a release yet. Without it the plugin is fairly
useless currently.

toonn committed:
bitlbee-facebook: Add note for removal of the patch

This TODO note should remind whoever bumps the package next to check
whether the patch can be dropped.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 112760` at a74e48c6 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/040f1598fc2d18194d955112205a6e0c">bitlbee-facebook</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/networking/instant-messengers/bitlbee-facebook/default.nix:28:3:
```
   |
28 |   meta = {
   |   ^
```
</li>
  </ul>
</details>

r-rmcgibbo commented:
Result of `nixpkgs-review pr 112760` at a74e48c6 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/bff39f86700f0e85ffec7f256397cdca">bitlbee-facebook</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/networking/instant-messengers/bitlbee-facebook/default.nix:28:3:
```
   |
28 |   meta = {
   |   ^
```
</li>
  </ul>
</details>

toonn committed:
bitlbee-facebook: Take on maintainership

SuperSandro2000 commented on pkgs/applications/networking/instant-messengers/bitlbee-facebook/default.nix:
```diff
@@ -12,6 +12,10 @@ stdenv.mkDerivation rec {
 
+  # TODO: This patch should be included with the next release after v1.2.1
+  #       these lines and the patch should be removed when this happens.
+  patches = [ ./0001-Bump-the-FB_ORCA_AGENT-version-once-again-208.patch ];
```
Can you use fetchpatch to fetch the two commits instead?

toonn replied:
Done. It's only a single commit, fyi. This doesn't include the base commit in the patch. Though I suppose it may not matter because `patch` probably ignores that anyway. Thought it might be human-friendlier though.

toonn committed:
bitlbee-facebook: Switch to fetchpatch

By using fetchpatch we avoid including another patch file in Nixpkgs.
One difference with the generated patch file is the base commit isn't
included in the patches from GitHub. The `patch` tool probably doesn't
look at these anyway so it might not matter.

SuperSandro2000 commented on pkgs/applications/networking/instant-messengers/bitlbee-facebook/default.nix:
```diff
@@ -12,6 +12,17 @@ stdenv.mkDerivation rec {
+    (fetchpatch {
+        name = "FB_ORCA_AGENT_version_bump.patch";
+        url = "https://github.com/bitlbee/bitlbee-facebook/commit/49ea312d98b0578b9b2c1ff759e2cfa820a41f4d.patch";
+        sha256 = "1c7ipbjg3560hg90zmg9r599287xfka1f7wxhiqzknikcjrcj8br";
```
```suggestion
        sha256 = "0nzyyg8pw4f2jcickcpxq7r2la5wgl7q6iz94lhzybrkhss5753d";
```

toonn replied:
It's not simply the result of `nix-prefetch-url` with the proper `--name=`? Thanks for the hash, don't have a linux box I want to build on rn.

SuperSandro2000 replied:
name does not change the hash. fetchpatch normalizes the patch which changes the hash.

toonn committed:
bitlbee-facebook: Correct patch hash

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

SuperSandro2000 commented on pkgs/applications/networking/instant-messengers/bitlbee-facebook/default.nix:
```diff
@@ -1,4 +1,4 @@
-{ lib, fetchFromGitHub, stdenv, bitlbee, autoconf, automake, libtool, pkg-config, glib, json-glib }:
+{ lib, fetchFromGitHub, fetchpatch, stdenv, bitlbee, autoconf, automake, libtool, pkg-config, glib, json-glib }:
 
 with lib;
```
```suggestion
```

toonn committed:
bitlbee-facebook: Remove broad "with lib"

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

SuperSandro2000 commented on pkgs/applications/networking/instant-messengers/bitlbee-facebook/default.nix:
```diff
@@ -1,6 +1,5 @@
-{ lib, fetchFromGitHub, stdenv, bitlbee, autoconf, automake, libtool, pkg-config, glib, json-glib }:
+{ lib, fetchFromGitHub, fetchpatch, stdenv, bitlbee, autoconf, automake, libtool, pkg-config, glib, json-glib }:
```
```suggestion
{ lib, fetchFromGitHub, fetchpatch, stdenv, bitlbee, autoconf, automake, libtool, pkg-config, json-glib }:
```

SuperSandro2000 committed:
Update pkgs/applications/networking/instant-messengers/bitlbee-facebook/default.nix
